pkg/agent/codemonkey/agent.go

```diff
@@ -0,0 +1,289 @@
+	"strings"
+
+	itools "github.com/Swarmind/libagent/internal/tools"
+	"github.com/Swarmind/libagent/pkg/config"
+	ptools "github.com/Swarmind/libagent/pkg/tools"
+
+	"github.com/tmc/langchaingo/llms/openai"
+	"github.com/rs/zerolog"
+)
+
+var log = zerolog.New(os.Stderr).With().Timestamp().Logger()
+
+// CodeMonkeyAgent implements the Code Monkey flow for automated issue resolution
+type CodeMonkeyAgent struct {
+	LLM           *openai.LLM
+	ToolsExecutor *itools.ToolsExecutor
+	WorkDir       string
+}
+
+// NewCodeMonkeyAgent creates a new Code Monkey Agent instance
+func NewCodeMonkeyAgent(ctx context.Context, cfg config.Config) (*CodeMonkeyAgent, error) {
+	llm, err := openai.New(
+		openai.WithBaseURL(cfg.AIURL),
+		openai.WithToken(cfg.AIToken),
+		openai.WithModel(cfg.Model),
+		openai.WithAPIVersion("v1"),
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create LLM: %w", err)
+	}
+
+	toolsExecutor, err := ptools.NewToolsExecutor(ctx, cfg, ptools.WithToolsWhitelist(
+		ptools.ReWOOToolDefinition.Name,
+		ptools.SemanticSearchDefinition.Name,
+		ptools.CommandExecutorDefinition.Name,
+		ptools.DDGSearchDefinition.Name,
+		ptools.WebReaderDefinition.Name,
+	))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create tools executor: %w", err)
+	}
+
+	workDir, err := os.MkdirTemp("", "codemonkey_*")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create work directory: %w", err)
+	}
+
+	return &CodeMonkeyAgent{
+		LLM:           llm,
+		ToolsExecutor: toolsExecutor,
+		WorkDir:       workDir,
+	}, nil
+}
+
+// executeGitCommand executes a git command using the command executor tool
+func (a *CodeMonkeyAgent) executeGitCommand(ctx context.Context, repoPath string, command string) error {
+	cmd := ptools.CommandExecutorArgs{
+		Command: fmt.Sprintf("cd %s && %s", repoPath, command),
+	}
+	cmdBytes, err := json.Marshal(cmd)
+	if err != nil {
+		return fmt.Errorf("failed to marshal command: %w", err)
+	}
+
+	_, err = a.ToolsExecutor.CallTool(ctx,
+		ptools.CommandExecutorDefinition.Name,
+		string(cmdBytes),
+	)
+	return err
+}
+
+// ResolveIssue implements the Code Monkey flow for issue resolution
+func (a *CodeMonkeyAgent) ResolveIssue(ctx context.Context, repoURL string, issueInput string) (string, error) {
+	// Step 1: Get issue text from GitHub
+	issueText, err := a.getIssueText(ctx, repoURL, issueInput)
+	if err != nil {
+		return "", fmt.Errorf("failed to get issue text: %w", err)
+	}
+
+	// Extract owner and repo from URL for PR creation
+	parts := strings.Split(strings.TrimSuffix(repoURL, ".git"), "/")
+	if len(parts) < 2 {
+		return "", fmt.Errorf("invalid repository URL: %s", repoURL)
+	}
+	owner := parts[len(parts)-2]
+	repo := parts[len(parts)-1]
+
+	// Step 2: Clone the repository
+	repoName := filepath.Base(strings.TrimSuffix(repoURL, ".git"))
+	repoPath := filepath.Join(a.WorkDir, repoName)
+
+	if err := a.executeGitCommand(ctx, a.WorkDir, fmt.Sprintf("git clone %s %s", repoURL, repoName)); err != nil {
```

Comment:
that should be a part of the generated rewoo plan
or, you can do the preparation work using go-github/go-git packages